cairo-contracts/packages/contracts/src/tests/transfer.cairo

```diff
@@ -148,3 +149,89 @@ fn test_mint_burn_roundtrip() {
     // Check the total supply of the ERC20 contract.
     erc20.assert_total_supply(0);
 }
+
+#[test]
+fn test_mint_with_pre_created_token() {
+    // -----------------------------------------------------------
+    // Setup Essentials
+    // -----------------------------------------------------------
+
+    let (core, ics20, _, _, _, transfer_cfg, mut spy) = setup(Mode::WithChannel);
+
+    let prefixed_denom = transfer_cfg.prefix_hosted_denom();
+
+    /// Pre-creates the IBC token
+    let token_address = ics20.create_ibc_token(prefixed_denom.clone());
+
+    // -----------------------------------------------------------
+    // Mint
+    // -----------------------------------------------------------
+
+    let msg_recv_packet = transfer_cfg
+        .dummy_msg_recv_packet(transfer_cfg.hosted_denom.clone(), COSMOS(), STARKNET());
+
+    core.recv_packet(msg_recv_packet.clone());
+
+    spy
+        .assert_recv_event(
+            ics20.address, CS_USER(), SN_USER(), prefixed_denom.clone(), transfer_cfg.amount, true,
+        );
+
+    let mut erc20: ERC20Contract = token_address.into();
+
+    spy.assert_transfer_event(erc20.address, ics20.address, SN_USER(), transfer_cfg.amount);
+
+    erc20.assert_balance(SN_USER(), transfer_cfg.amount);
+
+    erc20.assert_total_supply(transfer_cfg.amount);
+}
+
+#[test]
+fn test_create_ibc_token_ok() {
+    let (_, ics20, _, _, _, transfer_cfg, mut spy) = setup(Mode::WithChannel);
+    let prefixed_denom = transfer_cfg.prefix_hosted_denom();
+    let address = ics20.create_ibc_token(prefixed_denom.clone());
+    spy.assert_create_token_event(ics20.address, NAME(), SYMBOL(), DECIMAL_ZERO, address, 0);
+    let queried = ics20.ibc_token_address(prefixed_denom.key());
```

Comment:
reference to my last comment: essentially what you have done here in integration test, should be done in e2e test too.